### PR TITLE
LibAdapterBase ignore_files defaults to None

### PR DESCRIPTION
**Pull request by moellep**

### CODE REVIEW

**Comment by moellep** [comment]
Fixes #4688 lib ignore_files defaults to None

File: sirepo/lib.py [hunk 1]
```diff
@@ -22,11 +22,11 @@
 class LibAdapterBase:
     """Common functionality between code specific LibAdapter implementations."""
 
-    def __init__(self, ignore_files):
+    def __init__(self, ignore_files=None):
         m = inspect.getmodule(self)
         self._sim_data, _, self._schema = sirepo.sim_data.template_globals(m.SIM_TYPE)
         self._code_var = m.code_var
-        self._ignore_files = ignore_files
+        self._ignore_files = ignore_files if ignore_files else []
```
Comment (gurhar1133) [inline sirepo/lib.py:29]: simpler to do
```python
def __init__(self, ignore_files=[]):
    ...
    self.ignore_files = ignore_files
```
?

**Review by gurhar1133 — COMMENTED** [review]
Just one comment

File: sirepo/lib.py (hunk 1 above)
Comment (moellep) [reply sirepo/lib.py:29]: Yes, that is a python gotcha - you should never have an mutable as a default arg:
https://docs.python-guide.org/writing/gotchas/

Comment (gurhar1133) [reply sirepo/lib.py:29]: 👍

**Review by gurhar1133 — APPROVED** [review]

File: sirepo/lib.py (hunk 1 above)
Comment (e-carlin) [reply sirepo/lib.py:29]: You'll get surprising behavior using mutable default values for kwargs (lists are mutable). So, we don't use them.
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

**Pull request merged**